bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiHandlerFactory.java

```diff
@@ -45,21 +48,32 @@
  * @author Paul Frank - Initial contribution
  * @author Christoph Weitkamp - Improvements on channels for opening PVR TV or Radio streams
  */
+@NonNullByDefault
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.kodi")
 public class KodiHandlerFactory extends BaseThingHandlerFactory {
 
-    private Logger logger = LoggerFactory.getLogger(KodiHandlerFactory.class);
+    private final Logger logger = LoggerFactory.getLogger(KodiHandlerFactory.class);
 
-    private AudioHTTPServer audioHTTPServer;
-    private NetworkAddressService networkAddressService;
-    private WebSocketClient webSocketClient;
+    private final AudioHTTPServer audioHTTPServer;
+    private final NetworkAddressService networkAddressService;
+    private final KodiDynamicStateDescriptionProvider stateDescriptionProvider;
+    private final WebSocketClient webSocketClient;
 
-    // url (scheme+server+port) to use for playing notification sounds
-    private String callbackUrl = null;
+    private final Map<String, @Nullable ServiceRegistration<AudioSink>> audioSinkRegistrations = new ConcurrentHashMap<>();
 
-    private Map<String, ServiceRegistration<AudioSink>> audioSinkRegistrations = new ConcurrentHashMap<>();
+    // url (scheme+server+port) to use for playing notification sounds
+    private @Nullable String callbackUrl;
 
-    private KodiDynamicStateDescriptionProvider stateDescriptionProvider;
+    @Activate
+    public KodiHandlerFactory(final @Reference AudioHTTPServer audioHTTPServer,
+            final @Reference NetworkAddressService networkAddressService,
+            final @Reference KodiDynamicStateDescriptionProvider stateDescriptionProvider,
+            final @Reference WebSocketFactory webSocketFactory) {
+        this.audioHTTPServer = audioHTTPServer;
+        this.networkAddressService = networkAddressService;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.webSocketClient = webSocketFactory.getCommonWebSocketClient();
+    }
 
     @Override
     protected void activate(ComponentContext componentContext) {
```

Comment:
Please add `@Activate` annotation here too. Otherwise this method will be ignored in favor is all annotated activation methods.

Another option would be to pass the `ComponentContext` parameter to the constructor and move all the logic inside this method into the constructor too.

Reply:
Excuse me but I have a serious doubt. Please look at the handler factory for the Sonos and LGwebOS bindings for example. They are done like that. If `activate` method was not called, the LGwebOS binding would just not work at all a(s the WebSocket client would not be started).

Reply:
Hm, interesting. I do not have a clue why it is working but it seems to work.